rust/libnewsboat/src/utils.rs

```diff
@@ -58,6 +61,32 @@ pub fn to_u(rs_str: String, default_value: u32) -> u32 {
     result.unwrap()
 }
 
+/// Combine a base URL and a link to a new absolute URL.
+/// If the base URL is malformed or joining with the link fails, link will be returned.
+/// # Examples
+/// ```
+/// use libnewsboat::utils::absolute_url;
+/// assert_eq!(absolute_url("http://foobar/hello/crook/", "bar.html"),
+///     "http://foobar/hello/crook/bar.html".to_owned());
+/// assert_eq!(absolute_url("https://foobar/foo/", "/bar.html"),
+///     "https://foobar/bar.html".to_owned());
+/// assert_eq!(absolute_url("https://foobar/foo/", "http://quux/bar.html"),
+///     "http://quux/bar.html".to_owned());
+/// assert_eq!(absolute_url("http://foobar", "bla.html"),
+///     "http://foobar/bla.html".to_owned());
+/// assert_eq!(absolute_url("http://test:test@foobar:33", "bla2.html"),
+///     "http://test:test@foobar:33/bla2.html".to_owned());
+/// assert_eq!(absolute_url("foo", "bar"), "bar".to_owned());
+/// ```
+pub fn absolute_url(base_url: &str, link: &str) -> String {
```

Comment:
I was thinking if it would make sense eventually to return a `std::borow::Cow<str>` here to save the allocation in case `link` is returned. But we end up transforming it to a CString anyway which would then need an allocation, so it probably dosn't make much sense.

Reply:
We're committing a lot of performance sins already (e.g. converting `char*` to `String` only to borrow it later), so I think we can live with this one as well :) Really, I'm more concerned with correctness than performance right now, so this optimization can wait.